src/main/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsTransporter.java

```diff
@@ -0,0 +1,232 @@
+import java.util.Base64;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * This {@code SuccessFactorsTransporter} class is used to
+ * make a rest web service call to the SAP SuccessFactors exposed services.
+ */
+public class SuccessFactorsTransporter {
+  public static final String SERVICE_VERSION = "dataserviceversion";
+  private static final Logger LOG = LoggerFactory.getLogger(SuccessFactorsTransporter.class);
+  private static final long CONNECTION_TIMEOUT = 300;
+  private static final long WAIT_TIME = 5;
+  private static final long MAX_NUMBER_OF_RETRY_ATTEMPTS = 5;
+  private final String username;
+  private final String password;
+
+  public SuccessFactorsTransporter(String username, String password) {
+    this.username = username;
+    this.password = password;
+
+  }
+
+  /**
+   * Calls the Successfactors entity for the given URL and returns the respective response.
+   * Supported calls are:
+   * - testing the URL correctness
+   * - fetching the SuccessFactors entity metadata
+   * - fetching the total available records count
+   *
+   * @param endpoint  type of URL
+   * @param mediaType mediaType for Accept header property, supported types are 'application/json' & 'application/xml'
+   * @param fetchType type of call i.e. TEST / METADATA / COUNT, used for logging purpose.
+   * @return {@code SuccessFactorsResponseContainer}
+   * @throws TransportException any http client exceptions are wrapped under it
+   */
+  public SuccessFactorsResponseContainer callSuccessFactorsEntity(URL endpoint, String mediaType, String fetchType)
+    throws TransportException {
+
+    try {
+      LOG.debug(ResourceConstants.DEBUG_CALL_SERVICE_START.getMsgForKey(fetchType));
+      Response res = transport(endpoint, mediaType);
+      LOG.debug(ResourceConstants.DEBUG_CALL_SERVICE_END.getMsgForKey(fetchType));
+
+      return prepareResponseContainer(res);
+    } catch (IOException ioe) {
+      throw new TransportException(ResourceConstants.ERR_CALL_SERVICE_FAILURE.getMsgForKey(), ioe);
+    }
+  }
+
+  /**
+   * Calls the Successfactors entity to fetch the records with subsequent retries in case of failure.
+   * Retry modes are:
+   * - any HTTP code equal or above 500
+   * - max retry is 3 times
+   *
+   * @param endpoint record fetch URL
+   * @return {@code SuccessFactorsResponseContainer}
+   * @throws TransportException   any http client exceptions are wrapped under it
+   * @throws InterruptedException any error in case of retry suspension state
+   */
+  public SuccessFactorsResponseContainer callSuccessFactorsWithRetry(URL endpoint)
+    throws TransportException, InterruptedException, IOException {
+
+    LOG.debug(ResourceConstants.DEBUG_CALL_SERVICE_START.getMsgForKey(SuccessFactorsService.DATA));
+    Response res = retrySapTransportCall(endpoint, MediaType.APPLICATION_JSON);
```

Comment:
throw TransportException if retrySapTransportCall throws exception

Reply:
```
 public SuccessFactorsResponseContainer callSuccessFactorsWithRetry(URL endpoint)
    throws TransportException, IOException 
```

`TransportException` is beign already thrown. Removed `InterruptedException` 288b022add0e0a7a19652bd3d4efff4b18f3bf2f